test/common/ssl/ssl_socket_test.cc

```diff
@@ -550,8 +741,8 @@ TEST_P(SslSocketTest, GetNoUriWithDnsSan) {
 )EOF";
 
   // The SAN field only has DNS, expect "" for uriSanPeerCertificate().
-  testUtil(client_ctx_yaml, server_ctx_yaml, "", "", "", TEST_SAN_DNS_CERT_SERIAL, "", "", "",
-           "ssl.handshake", true, GetParam());
+  TestUtilOptions test_options(client_ctx_yaml, server_ctx_yaml, "ssl.handshake", true, GetParam());
```

Comment:
There's still 5 params here, which is getting a little long, could some of these be switched to the builder pattern, or are they all compulsory?

Reply:
@htuch I can switch all the parameters to be set by setters, however it will produce longer code at the call sites. How about leaving four parameters in the constructor, for TestUtilOptions and TestUtilOptionsV2? It will be client ctx, server ctx, expected_success and version. "ssl.handshake" can be set by default for client and server stats.

Reply:
SGTM, thanks.